libs/core/src/lib/fixed-card-layout/fixed-card-layout.component.ts

```diff
@@ -273,66 +392,51 @@ export class FixedCardLayoutComponent implements OnInit, AfterContentInit, After
-        const availableLayoutWidth = this.getWidthAvailable();
-        const numberOfCardsWithNoGap = Math.floor(availableLayoutWidth / this.cardMinimumWidth);
-        const requiredWidthWithGap =
-            numberOfCardsWithNoGap * this.cardMinimumWidth + (numberOfCardsWithNoGap - 1) * CARD_GAP_WIDTH;
+    /** @hidden */
+    private _processCards(): void {
+        this._cardsArray = this._cards
+            .toArray()
+            .sort((firstCard, secondCard) => firstCard.fdCardDef - secondCard.fdCardDef);
 
-        if (requiredWidthWithGap > availableLayoutWidth) {
-            columnCount = numberOfCardsWithNoGap - 1;
-        } else {
-            columnCount = numberOfCardsWithNoGap;
-        }
+        this._changeDetector.detectChanges();
 
-        // minimum number of column to 1
-        columnCount = columnCount ? columnCount : 1;
-        this.layoutChange.emit(new Layout(columnCount, availableLayoutWidth));
-        return columnCount;
+        this.updateLayout();
     }
 
-    /**
-     * @hidden Renders layout on column changes.
-     */
-    private _renderLayout(): void {
-        // convert latest cards queryList to Array of cards
-        this._cardsArray = this.cards.toArray();
-        this._initializeColumns(this._numberOfColumns);
-        this._distributeCards(this._columns);
-        this._changeDetector.detectChanges();
-    }
+    /** @hidden Distribute cards among columns to arrange them in "Z" flow */
+    private _setCardColumns(): void {
+        const columns: CardColumn[] = new Array(this._numberOfColumns).fill(0).map(() => new Array(0));
 
-    /**
-     * @hidden Initialize columns with empty arrays
-     */
-    private _initializeColumns(numberOfColumns: number): void {
-        this._columns = [];
-        for (let i = 0; i < numberOfColumns; i++) {
-            this._columns.push([]);
-        }
+        this._cardsArray.forEach((card, i) => columns[i % this._numberOfColumns].push(card));
+
+        this._cardColumns = columns;
     }
+}
 
-    /**
-     * @hidden Redistribute cards among columns
-     */
-    private _distributeCards(columns: CardColumn[]): void {
-        const numberOfColumns = columns.length;
-        // sort cards based on rank then create layout
-        this._cardsArray?.sort(comparator);
-        this._cardsArray?.forEach((card, i) => {
-            const index = i % numberOfColumns;
-            columns[index].push(card);
-        });
+/** @hidden Returns number of columns that can fit in current available width for fd-card-layout */
+function getNumberOfColumns(containerWidth: number, cardWidth: number): number {
+    const numberOfCardsWithNoGap = Math.floor(containerWidth / cardWidth); // 960 / 320 = 3
+    const requiredWidthWithGap = numberOfCardsWithNoGap * cardWidth + (numberOfCardsWithNoGap - 1) * CARD_GAP_WIDTH;
+    const columnCount = requiredWidthWithGap > containerWidth ? numberOfCardsWithNoGap - 1 : numberOfCardsWithNoGap;
```

Comment:
this is quite unlikely to happen, but if containerWidth or cardWidth is 0, or cardWidth > containerWidth, we will have columnCount < 0 or columnCount === Infinity. 
I think with these changes it will be slightly safer
```typescript
function getNumberOfColumns(containerWidth: number, cardWidth: number): number {
    containerWidth ||= 1;
    cardWidth ||= 1;
    const numberOfCardsWithNoGap = Math.max(Math.floor(containerWidth / cardWidth), 1); // 960 / 320 = 3
    const requiredWidthWithGap = numberOfCardsWithNoGap * cardWidth + (numberOfCardsWithNoGap - 1) * CARD_GAP_WIDTH;
    const columnCount = requiredWidthWithGap > containerWidth ? numberOfCardsWithNoGap - 1 : numberOfCardsWithNoGap;
    return Math.max(columnCount, 1)
}
```

Also can you please add unit tests for it?